### ISSUE TITLE
Some UI elements are missing accessibility labels

### ISSUE THREAD

**Issue by jackbentley** [pradyunsg/furo#93]
**Is your feature request related to a problem? Please describe.**
Accessibility for users is becoming more and more of a key thing for websites, projects, and companies.

Section 508, CVAA, ADA, Equality Act, Public Sector Bodies Accessibility Regulations, etc, etc.

**Describe the solution you'd like**
The template should pass accessibility tests in accordance with WCAG 2.1 AA out of the box.

Specifically, there are currently some issues with colour contrast, the search form (labels, submit button), and sidebar expand/collapse buttons having no labels.

Basic testing can be achieved with tools like [WAVE](https://wave.webaim.org/), [SiteImprove Accessibility Checker](https://chrome.google.com/webstore/detail/siteimprove-accessibility/efcfolpjihicnikpmhnmphjhhpiclljc), and [axe](https://www.deque.com/axe/browser-extensions/). There is the possibility of integrating tools for automated testing of most things, also.

**Describe alternatives you've considered**
N/A.

**Additional context**
Any user changes to the theme which break accessibility are obviously out-of-scope. Ensuring changes are accessible would be up to the user implementing the changes.

**Comment by pradyunsg** [comment]
PRs welcome!

**Comment by pradyunsg** [comment]
Looked into this yesterday evening.

I think all the labels that Wave and Chrome's Lighthouse used to complain about have been added. For anyone who wants to see the changes, they're all in #208 and https://github.com/pradyunsg/furo/compare/42b4cde2de15bf2ee5da11f1dda8b6e13a61262f..44ad20ef2ad76c83f3e33da07e83c1c2c84bb942.

I'll cut a release over the weekend.
